README.MD

```diff
@@ -0,0 +1,326 @@
+let gatewayAddress;
+mockContractDeployerObject.deployMockGatewayPass().then(function(){
+      gatewayAddress = mockContractDeployerObject.addresses.MockGatewayPass;
+});
+
+```
+
+#### BrandedToken Staking through GatewayComposer
+
+##### Approve GatewayComposer for ValueToken & Calls requestStake
+
+```js
+const StakeHelper = BrandedToken.Helpers.StakeHelper;
+const stakeHelperInstance = new StakeHelper(web3Provider, brandedTokenAddress, gatewayComposerAddress);
+let txOptions = {
+  from: owner
+};
+const stakeAmountInWei = '100';
+const stakerGatewayNonce = 1;
+let mintBTAmountInWei;
+stakeHelperInstance.convertToBTToken(stakeAmountInWei, brandedTokenAddress, web3Provider, txOptions).then(function(result){
+    mintBTAmountInWei = result;
+});
+const stakerInstance = new Staker(web3Provider, eip20Token, brandedTokenAddress, gatewayComposerAddress);
+const stakeGasPrice = '0x3B9ACA00';
+const stakeGasLimit = '7500000';
+const beneficiary = '0xaabb1122....................';
+stakerInstance.requestStake(
+    mockTokenAbi,
+    owner,
+    stakeAmountInWei,
+    mintBTAmountInWei,
+    gatewayAddress,
+    stakeGasPrice,
+    stakeGasLimit,
+    beneficiary,
+    stakerGatewayNonce,
+    txOptions
+).then(function(){
+  console.log("Request stake completed!");
+})
+
+```           
+
+##### Facilitator collects worker signature
+The facilitator is a service that facilitates progress of the stake and mint.
+```js
+// StakeRequest nonce stored in BrandedToken contract.
+let stakeRequestNonce;
+// 1. Create TypedData
+const stakeRequestTypedData = stakeHelperInstance.getStakeRequestTypedData(stakeAmountInWei, stakeRequestNonce);
+
+// 2. Generate EIP712 signature object.
+const workerAccountInstance = web3Provider.eth.accounts.wallet[worker];
+workerAccountInstance.signEIP712TypedData(stakeRequestTypedData).then(function(signature){
+  console.log("signature:", signature);
+})
+
+```
+
+##### Facilitator calls acceptStakeRequest
+
+```js
+const Mosaic = require('@openstfoundation/mosaic');
+const hashLockInstance = Mosaic.Helpers.StakeHelper.createSecretHashLock();
+const gatewayContractInstance = Mosaic.Contracts.getEIP20Gateway(web3Provider, gatewayAddress);
+let bountyAmountInWei;
+gatewayContractInstance.methods.bounty().call().then(function(result){
+  bountyAmountInWei = result;
+});
+const Facilitator = BrandedToken.Helpers.Facilitator;
+const facilitatorInstance = new Facilitator(web3Provider, eip20Token, brandedTokenAddress, gatewayComposerAddress, facilitator);
+txOptions = {
+  gas: '7500000'
+};    
+let stakeRequestHash; // stakeRequestHash unique per stake request stored in GatewayComposer.stakeRequests mapping
+facilitatorInstance.acceptStakeRequest(
+  stakeRequestHash,
+  signature,
+  bountyAmountInWei,
+  mockTokenAbi,
+  hashLockInstance.hashLock,
+  web3Provider,
+  txOptions
+).then(function(){
+  console.log("Successfully completed acceptStakeRequest!");
+})
+```
+
+#### Follow below steps to run integration tests locally
```

Comment:
Suggest replacing this section with the following (the heading level notwithstanding):

#### Tests

Tests require docker-compose. To run the tests, execute below command from root directory.

 ```bash
    npm run test
 ```

Reply:
Adjusted.